src/test/java/org/apache/sysds/test/AutomatedTestBase.java

```diff
@@ -1380,80 +1386,116 @@ public static int getRandomAvailablePort() {
 			return availableSocket.getLocalPort();
 		}
 		catch(IOException e) {
-			// If no port was found just use 9999
+			// If no port was found just use 9990
 			return 9990;
 		}
```

Comment:
i think i did this, because i experienced weird behavior in cases where i searched for a random available port recursively, and i did not want a potential infinite loop in our tests startup.
But i do agree that searching for a valid port would be better, than just returning this default.

Reply:
If you get an IOException upon creating a socket, there are probably more severe issues pending than finding a nice port number :fearful: